pywbem/cim_obj.py

```diff
@@ -2511,6 +2511,72 @@ def to_wbem_uri(self, format='standard'):
 
         return _ensure_unicode(''.join(ret))
 
+    @staticmethod
+    def from_instance(class_, instance, namespace=None, host=None,
+                      strict=False):
+        """
+        Given a class and corresponding instance, create and return the
+        CIMInstanceName from the class key properties and instance key
+        property values.
```

Comment:
It is sufficient that the class is a superclass of the creation class of the instance, as long as it has all key properties defined.
Actually, that is a requirement we need to state explicitly, because it may be that the `CIMClass` object was created with `LocalOnly=True` and therefore has only its additional property definitions, which may not even include any key property.

Reply:
We cannot know that at this time since we do not have access to a repository. Therefore, we either ignore the issue about the relation between the classs except for documentation or keep the existing test which expects the classname to be the same as the instancename.

**DISCUSSION:** Should we drop the test for classname matching classname in instance??

Reply:
I think we should drop the test.
And yes, I meant the requirement only to be documented, but not to be implemented as a check.